src/ecpair.js

```diff
@@ -58,20 +58,24 @@ ECPair.fromPublicKeyBuffer = function (buffer, network) {
 }
 
 ECPair.fromWIF = function (string, network) {
-  var buffer = bs58check.decode(string)
+  network = network || NETWORKS.bitcoin
+  var decoded = wif.decode(string)
+  var version = decoded.version
 
+  // [network, ...]
   if (types.Array(network)) {
-    var version = buffer[0]
-
     network = network.filter(function (network) {
       return version === network.wif
     }).pop()
 
     if (!network) throw new Error('Unknown network version')
+
+  // network
+  } else if (network) {
```

Comment:
`network` always will be truthly here?
why not `} else if (version !== network.wif) throw new Error('Invalid network version')` ?